docs/source/API/core/builtinreducers/ReducerConcept.rst

```diff
@@ -92,6 +92,22 @@ Functions
 
     * Returns a view of the result place.
 
+Requirements
+~~~~~~~~~~~~
+
+The reducer is assumed to define a commutative monoid with respect to the value type it is used with, i.e., the binary operation
+
+.. code-block:: cpp
+
+    value_type op(const value_type& val1, const value_type& val2) {
+      value_type result = val1;
+      reducer.join(result, val2);
+      return result;
+    }
+
+is commutative and associative with identity element that can be set by calling ``reducer.init(el)``.
+
```

Comment:
We could also go with a "non-mathy" explanation:
```c++
auto foo(auto reducer, auto val1, auto val2) {
   auto result = val1;
   reducer.join(result, val2);
   return result;
}
```

returns `val` for both `foo(reducer, init, val)` and `foo(reducer, val, init)` where `init` is a value set by calling `reducer.init(init)`.

Reply:
I think your suggestion is a bit more appropriate for the wiki.

Reply:
Do you want to open a PR for it?